src/node/SetupServerApi.ts

```diff
@@ -29,6 +34,10 @@ export type InterceptorsList = Array<{
   new (): Interceptor<HttpRequestEventMap>
 }>
 
+interface InteractiveRequest extends Request {
+  respondWith(response: Response): void
+}
```

Comment:
Here I created an interface to allow extracting a (private) function for the response. I couldn't find any exported type, but if you wanted to export from the interceptors package this could be removed and replaced with an import?

Alternatively, the private function can be removed and the code duplicated, or there may be alternatives for bypassing.

Reply:
Removed in https://github.com/mswjs/msw/pull/2028/commits/64751b3184830834af988b61695b5cdf8a234479 since the same result can be achieved without the private function or duplication (early return produces desired behaviour).